### PR TITLE
Fix setting record with default values

### PR DESCRIPTION
**Pull request by darh**
Please take a look.

### CODE REVIEW

File: compose/service/record_test.go [hunk 1]
```diff
@@ -297,9 +297,167 @@ func TestRecord_boolFieldPermissionIssueKBR(t *testing.T) {
...
 		req.Equal("1", recChecked.Values.Get("bool", 0).Value, "should checked again")
+	}
+}
+
+func TestRecord_defValueFieldPermissionIssue(t *testing.T) {
+	var (
+		req = require.New(t)
+
+		// uncomment to enable sql conn debugging
+		//ctx = logger.ContextWithValue(context.Background(), logger.MakeDebugLogger())
```
Comment (tjerman) [inline compose/service/record_test.go]: leftover

File: compose/service/record_test.go [hunk 2]
```diff
@@ -297,9 +297,167 @@ func TestRecord_boolFieldPermissionIssueKBR(t *testing.T) {
...
+
+		recPartial *types.Record
+
+		//valChecked = types.RecordValueSet{
+		//	&types.RecordValue{Name: "string", Value: "abc"},
+		//}
+		//
+		//valUnchecked = types.RecordValueSet{
+		//	&types.RecordValue{Name: "string", Value: "abc"},
+		//}
```
Comment (tjerman) [inline compose/service/record_test.go]: rm

File: compose/service/record_test.go [hunk 3]
```diff
@@ -297,9 +297,167 @@ func TestRecord_boolFieldPermissionIssueKBR(t *testing.T) {
...
+					out += "NULL"
+				}
+				out += ">"
+			}
+
+			return
+		}
+	)
+
+	t.Log("creating namespace, module and fields")
```
Comment (tjerman) [inline compose/service/record_test.go:361]: are these logs something we want to keep?

File: compose/service/record.go [hunk 4]
```diff
@@ -620,6 +620,30 @@ func RecordUpdateOwner(invokerID uint64, r, old *types.Record) *types.Record {
 	return r
 }
 
+func RecordValueCreateOpCheck(ctx context.Context, ac recordValueAccessController, m *types.Module, vv types.RecordValueSet) *types.RecordValueErrorSet {
```
Comment (tjerman) [inline compose/service/record.go]: This is unused as far as I can tell?

File: compose/service/record_test.go [hunk 5]
```diff
@@ -297,9 +297,167 @@ func TestRecord_boolFieldPermissionIssueKBR(t *testing.T) {
...
+
+		recPartial.Values = types.RecordValueSet{&types.RecordValue{Name: "writable", Value: "w2"}}
+
+		recPartial, err = svc.Update(ctx, recPartial)
+		req.NoError(errors.Unwrap(err))
+		req.Equal("<w2><NULL>", valueExtractor(recPartial, "writable", "readable"))
 	}
 
+	{
+		t.Log("creating record with w/o editor role (expecting defaults")
```
Comment (tjerman) [inline compose/service/record_test.go]: remove `w/o`

File: compose/service/record_test.go [hunk 6]
```diff
@@ -297,9 +297,167 @@ func TestRecord_boolFieldPermissionIssueKBR(t *testing.T) {
...
+		t.Log("updating record removing one of the values")
+
+		recPartial.Values = types.RecordValueSet{
+			&types.RecordValue{Name: "writable", Value: "def-w"},
+			&types.RecordValue{Name: "readable", Value: "r"},
+		}
+
+		recPartial, err = svc.Update(ctx, recPartial)
+		req.NoError(errors.Unwrap(err))
+		req.Equal("<def-w><r>", valueExtractor(recPartial, "writable", "readable"))
```
Comment (tjerman) [inline compose/service/record_test.go]: No value is removed here

**Review by tjerman — CHANGES_REQUESTED** [review]
Why are we doing a `t.Log("...test message details here...")` instead of `t.Run("...", ...)`?

**Comment by darh** [comment]
> Why are we doing a `t.Log("...test message details here...")` instead of `t.Run("...", ...)`?

because I'm telling a story of a complex test case and not trying to run test in parallel.

**Pull request closed without merging**